webapp/blog/views.py

```diff
@@ -19,22 +23,70 @@ def snap_posts(snap):
         blog_articles = None
         articles = []
 
+        third_party_blogs = get_yaml("blog/content/blog-posts.yaml")
+
+        if snap in third_party_blogs:
+            post = third_party_blogs[snap]
+            cdn_image = "/".join(
+                [
+                    "https://res.cloudinary.com",
+                    "canonical",
+                    "image",
+                    "fetch",
+                    "f_auto,q_auto,fl_sanitize,w_346,h_231,c_fill",
+                    post["image"],
+                ]
+            )
+            brand_image = "https://assets.ubuntu.com/v1/aae0f33a-omgubuntu.svg"
+            image_element = "".join(
+                [
+                    f'<img src="{cdn_image}" ',
+                    'style="display:block">',
+                    f'<img src="{brand_image}" ',
+                    'class="p-blog-post__source" />',
+                ]
+            )
+            articles.append(
+                {
+                    "slug": post["uri"],
+                    "title": post["title"],
+                    "image": image_element,
+                }
+            )
+
         if blog_tags:
             try:
                 blog_articles, total_pages = wordpress_api.get_articles(
-                    blog_tags["id"], 3
+                    blog_tags["id"], 3 - len(articles)
                 )
             except Exception:
                 blog_articles = []
 
             for article in blog_articles:
+                try:
+                    featured_media = wordpress_api.get_media(
+                        article["featured_media"]
+                    )["source_url"]
+                    featured_media = image_template(
+                        url=featured_media,
+                        alt="",
+                        width="346",
+                        height="231",
+                        hi_def=True,
+                        loading="auto",
+                    )
+                except Exception:
```

Comment:
Just a small comment: With `except Exception` you are covering anything that could fail in that code, so that means we could not get sentry notifications if something is wrong there and it is not what we expect.

If you know the specific exception it will be better to use it.

Reply:
I have seen that we are doing it also a few lines before for the other API call... So you can ignore me if you want

Reply:
Maybe both should be reviewed at some point?

Reply:
https://github.com/canonical-web-and-design/snapcraft.io/issues/2637